### PR TITLE
update spine to 3.8.95+

### PR DESCRIPTION
**Pull request by PatriceJiang**
re https://forum.cocos.org/t/cocos-creator-runtime-animation-timing-is-linear-instead-of-bezier/93518

upgrade spine to latest version.

### CODE REVIEW

File: extensions/spine/lib/spine.d.ts [hunk 1]
```diff
@@ -1,1331 +1,1337 @@
...
-		static ROTATE: number;
-		static TRANSLATE: number;
-		pathConstraintIndex: number;
-		frames: ArrayLike<number>;
-		constructor(frameCount: number);
-		getPropertyId(): number;
-		setFrame(frameIndex: number, time: number, rotateMix: number, translateMix: number): void;
-		apply(skeleton: Skeleton, lastTime: number, time: number, firedEvents: Array<Event>, alpha: number, blend: MixBlend, direction: MixDirection): void;
-	}
+declare module spine {
```
Comment (jareguo) [inline extensions/spine/lib/spine.d.ts]: 这里可能得手动改一下？我们是把 spine 暴露在 sp 这个全局变量下才对。之前应该是宛君自己有调整过。

Comment (PatriceJiang) [inline extensions/spine/lib/spine.d.ts]:
```suggestion
declare namespace sp.spine {
```

File: extensions/spine/lib/spine.d.ts [hunk 2]
```diff
@@ -1,1331 +1,1337 @@
...
-	}
-	abstract class AnimationStateAdapter implements AnimationStateListener {
-		start(entry: TrackEntry): void;
-		interrupt(entry: TrackEntry): void;
-		end(entry: TrackEntry): void;
-		dispose(entry: TrackEntry): void;
-		complete(entry: TrackEntry): void;
-		event(entry: TrackEntry, event: Event): void;
-	}
+declare module spine {
```
Comment (PatriceJiang) [inline extensions/spine/lib/spine.d.ts:247]:
```suggestion
declare namespace sp.spine {
```

File: extensions/spine/lib/spine.d.ts [hunk 3]
```diff
@@ -1,1331 +1,1337 @@
...
-	class AnimationStateData {
-		skeletonData: SkeletonData;
-		animationToMixTime: Map<number>;
-		defaultMix: number;
-		constructor(skeletonData: SkeletonData);
-		setMix(fromName: string, toName: string, duration: number): void;
-		setMixWith(from: Animation, to: Animation, duration: number): void;
-		getMix(from: Animation, to: Animation): number;
-	}
+declare module spine {
```
Comment (PatriceJiang) [inline extensions/spine/lib/spine.d.ts]:
```suggestion
declare namespace sp.spine {
```

File: extensions/spine/lib/spine.d.ts [hunk 4]
```diff
@@ -1,1331 +1,1337 @@
...
-		remove(path: string): void;
-		removeAll(): void;
-		isLoadingComplete(): boolean;
-		getToLoad(): number;
-		getLoaded(): number;
-		dispose(): void;
-		hasErrors(): boolean;
-		getErrors(): Map<string>;
-	}
+declare module spine {
```
Comment (PatriceJiang) [inline extensions/spine/lib/spine.d.ts:383]:
```suggestion
declare namespace sp.spine {
```

File: extensions/spine/lib/spine.d.ts [hunk 5]
```diff
@@ -1,1331 +1,1337 @@
...
-		atlas: TextureAtlas;
-		constructor(atlas: TextureAtlas);
-		newRegionAttachment(skin: Skin, name: string, path: string): RegionAttachment;
-		newMeshAttachment(skin: Skin, name: string, path: string): MeshAttachment;
-		newBoundingBoxAttachment(skin: Skin, name: string): BoundingBoxAttachment;
-		newPathAttachment(skin: Skin, name: string): PathAttachment;
-		newPointAttachment(skin: Skin, name: string): PointAttachment;
-		newClippingAttachment(skin: Skin, name: string): ClippingAttachment;
-	}
+declare module spine {
```
Comment (PatriceJiang) [inline extensions/spine/lib/spine.d.ts:411]:
```suggestion
declare namespace sp.spine {
```

File: extensions/spine/lib/spine.d.ts (hunk 1 above)
Comment (PatriceJiang) [reply extensions/spine/lib/spine.d.ts]: 已修改

**Review by jareguo — APPROVED** [review]

**Comment by balazsnemeth** [comment]
@jareguo , @PatriceJiang , do you plan to release this in v2.4.0? Do you have a release ETA?

**Comment by jareguo** [comment]
Hi @balazsnemeth , yes we are planing. Now you can download v2.4-preview from Cocos Dashboard, the dashboard can be downloaded from https://cocos.com/creator

**Comment by PatriceJiang** [comment]
This will break compatibility with older versions of spine files

**Comment by balazsnemeth** [comment]
@PatriceJiang , @jareguo, meaning it will be no Spine runtime update in v2.4.0? So we cannot use the features of Spine 3.8 in CocosCreator, but we need to stick with Spine 3.7? Please, confirm.

**Comment by jareguo** [comment]
Hi, you can still use Spine 3.8. We just cancelled the 3.8.95. If you need to use the feature you can still merge this PR in your customized engine.

**Comment by balazsnemeth** [comment]
@jareguo , may you know, was there any Spine version update on Spine version for CocosCreator 2.4.4? Regarding Spine 3.8, which minor version is the latest which is compatible  (because Spine 3.8.95 and above is not supported).

**Pull request closed without merging**